lib/hook.js

```diff
@@ -54,19 +59,23 @@ module.exports = function jsonApiHook(sails) {
       before: {
 
         'all /*': function (req, res, next) {
+          if (req.isSocket) return next();
           addResponseMethods(req, res);
           next();
         },
 
         'GET /*': function (req, res, next) {
+          if (req.isSocket) return next();
           addResponseMethods(req, res);
           normalizeQueryParams(req, res);
           next();
         },
 
         'POST /*': function (req, res, next) {
+          if (req.isSocket) return next();
           addResponseMethods(req, res);
           normalizePayload(req, next);
+          next();
```

Comment:
There shouldn't be a call to `next` here. Has to wait for `normalizePayload` and therefore that one should call `next`. After removing this line (and ignoring still present coding stile issues) all tests passed for 89212743a6ae4f17340fbbb8985c0671ec64e79c.